auth/auth.go

```diff
@@ -12,17 +14,47 @@ import (
 	"github.com/open-boardgame-stats/backend/ent"
 	"github.com/open-boardgame-stats/backend/ent/user"
 	"golang.org/x/crypto/bcrypt"
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/google"
 )
 
+const oAuthStateTTL = 20 * time.Minute
+
 type AuthService struct {
-	client *ent.Client
-	ctx    context.Context
-	secret string
+	client      *ent.Client
+	ctx         context.Context
+	secret      string
+	oAuthConfig oAuthConfig
+}
+
+type oAuthConfig struct {
+	google oauth2.Config
+}
+
+func NewOAuthConfig(serverHost, serverPort, googleClientID, googleClientSecret string) oAuthConfig {
+	return oAuthConfig{
+		google: oauth2.Config{
+			ClientID:     googleClientID,
+			ClientSecret: googleClientSecret,
+			Endpoint:     google.Endpoint,
+			RedirectURL:  fmt.Sprintf("http://%s:%s/auth/google/callback", serverHost, serverPort),
+			Scopes:       []string{"https://www.googleapis.com/auth/userinfo.email"},
```

Comment:
Let's also add the profile scope to get the user's name

Reply:
done